### ISSUE TITLE
Replay Level back to you and others

### ISSUE THREAD

**Issue by shoc71** [shoc71/dont-stop-now#248]
- replay to view myself beat the level
- export it to other players in a file
- have been a lil cyan half-transparent dude
- make some failed attempts replayable (last 5)
- make it race-able with friends

**Comment by slysnivy** [comment]
For the "make some failed attempts replayable" part, would that record the player going through the level including the latest 5 deaths?

**Comment by shoc71** [comment]
yes

**Comment by slysnivy** [comment]
For the file import/export part, are we expecting users to drag, drop and send files to specific file locations? It might be too complex depending on the audience we target. Therefore, we need a simpler way to do it. Let me know if you got any ideas, and I'll post some of my own as well.

**Comment by slysnivy** [comment]
For now, we can use text/strings instead of files where players can copy and paste the data through their own means (email/messages). 

If we want to use files, then we can experiment with this:

https://stackoverflow.com/questions/74407954/drag-drop-files-to-pygame-and-save-them

But it seems like it has limited usability and will be harder to implement. 

Of course, these are only temporary solutions for now and we'll think about better methods once the game is finished.

**Comment by shoc71** [comment]
> For the file import/export part, are we expecting users to drag, drop and send files to specific file locations? It might be too complex depending on the audience we target.

- I agree this may be too complicated for certain demographics but for now, that's our default mode. One idea that I had was to create a unique code that would code and decode the replay and show it perfectly but that would be too complicated.

**Comment by slysnivy** [comment]
So we're sticking with the file output/input?

**Comment by shoc71** [comment]
I'm thinking we should both options available fro players

**Comment by slysnivy** [comment]
So both text and file input/output

**Comment by shoc71** [comment]
yes

**Comment by slysnivy** [comment]
Ok for now, I'll implement the file io without any convenient features. Then we can think about that and the text later.

**Comment by shoc71** [comment]
sounds good, we'll resolve this issue when we get the text thing working

**Comment by shoc71** [comment]
> For text, would it be ok to do it by a level basis. Like one copy/paste code would be for one level.

Short answer - yes, keep it as a level basis thing for one level only.
That's what I was thinking of, as well. However, you reminded me that it's possible (although really hard; we likely won't do this), to export the entire gameplay.

**Comment by shoc71** [comment]
that sounds greeat

**Comment by slysnivy** [comment]
Should we have winning the replays based on who reaches the goal first (timed)?

**Comment by slysnivy** [comment]
So both players have to reach the goal for the level/race to be over (or when the player decides to quit and return to the menu)

**Comment by shoc71** [comment]
I'm imagining two separate functions happening here. Either the player is in replay mode or the player is playing the game. The two can not overlap with each other.

As for 
> Should we have winning the replays based on who reaches the goal first (timed)?
- Think it's fine just to record the latest instance of that character's victory (+5 previous attempts) and if there are any other prior victories they have achieved in the same level, may this be overwritten with the new one over the old one. (this is something we'll patch sometime later).

**Comment by slysnivy** [comment]
That's good to know, but my question was about playing the replays.

**Comment by slysnivy** [comment]
For now, I think I'll have it so the level ends once either the player or replay square reaches the end. So it'll be faster than what I mentioned earlier.

**Comment by shoc71** [comment]
Having a separate option screen (like pressing "g" for the input) for the level replay of someone else, think we might need to use tkinter for this because pygame might be messy.

**Comment by slysnivy** [comment]
That won't be necessary, as right now it has it's own option on MenuScene leading to it's own level selection screen (thanks to how we've got the classes set up).

**Comment by shoc71** [comment]
Oh, that sounds great and amazing!

**Comment by slysnivy** [comment]
I will also add a countdown for the replays (since in some cases, the replay race begins instantly)
